### DISCUSSION TITLE
Datepicker UI not visible/accessible when form is inside a <dialog>

### DISCUSSION THREAD

**Discussion by vonscriptor** [solspace/craft-freeform#1346] [Questions & Answers] [answered]
### What happened?

I have a form rendered in a `<dialog>` (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog) and the datepicker UI is not visible due to the dialog being in the `top layer` of the DOM (https://developer.mozilla.org/en-US/docs/Glossary/Top_layer) and `flatpickr-calendar` living outside of the dialog/form whose z-index then has no effect.

### How can we reproduce this?

1. Add a Date & Time field to a form inside of a `<dialog>`
2. Render form
3. See that the datepicker UI is not visible/accessible

### Versions

Freeform Pro 5.3.2
Craft Pro 4.9.6

**Comment by vonscriptor** [comment]
Can `flatpickr` mark-up, styles, and scripts all be injected inside of the form?

**Comment by kjmartens** [comment, marked as answer]
Sorry for the delay @vonscriptor,

This is a known issue with Flatpickr and `<dialog>`. However, you can workaround this by setting Flatpickr to `static: true`. To do this in Freeform, just add the following JS to your page somewhere: 

``` html
<script>
    document.addEventListener("flatpickr-before-init", (event) => {
        event.options.static = true;
    })
</script>
```

I tested this now to confirm that it does work correctly when used inside of a `<dialog>`. 🙂
